lib/map/security_token/service.rb

```diff
@@ -57,11 +57,17 @@ def parse_and_raise_error(e, icn, application)
 
       def parse_response(response, application, icn)
         response_body = response.body
+        if response_body['expires_in'].to_i > config.max_token_duration
+          raise Errors::InvalidTokenDurationError,
+                "#{config.logging_prefix} token failed, token duration exceeds maximum"
+        end
 
         {
           access_token: response_body['access_token'],
           expiration: Time.zone.now + response_body['expires_in']
         }
+      rescue Errors::InvalidTokenDurationError => e
+        raise e
```

Comment:
I dont think you need to rescue and re-raise here. This should get rescued in the `token` method